In [2]:
import pandas as pd
import json

In [3]:
with open(r"/Users/tiffanymarks/DataInEMAT22110/Instagram miffanytarks_20220906/likes/liked_posts.json") as t:
    insta_likes = json.load(t)

**a) Who collected this data? (we downloaded the data, but did we collect it?)**

The data comes from the Instagram application. Instagram is owned by **Meta Platforms, Inc.**, so Meta collected the data. 

In [4]:
likes = pd.DataFrame(insta_likes)

In [5]:
likes.head()

likes_media_likes
0  {'title': 'kristeacafe', 'media_list_data': []...
1  {'title': 'rakib.uiux', 'media_list_data': [],...
2  {'title': '3dgamingsetups', 'media_list_data':...
3  {'title': 'websitekana', 'media_list_data': []...
4  {'title': 'redheadstacyx', 'media_list_data': ...

I took the information from liked_post.json and transformed it in a DataFrame

In [6]:
separate_likes = pd.DataFrame(insta_likes['likes_media_likes'])

In [7]:
separate_likes.head()

title media_list_data  \
0     kristeacafe              []   
1      rakib.uiux              []   
2  3dgamingsetups              []   
3     websitekana              []   
4   redheadstacyx              []   

                                    string_list_data  
0  [{'href': 'https://www.instagram.com/reel/CcBC...  
1  [{'href': 'https://www.instagram.com/p/CczbkJ9...  
2  [{'href': 'https://www.instagram.com/p/Cc4H2pK...  
3  [{'href': 'https://www.instagram.com/p/Cc-Cn7I...  
4  [{'href': 'https://www.instagram.com/reel/CdDJ...

I took the likes_media_likes information and used pd.DataFrame to organized it into 3 columns of information

**b) Why was this data collected? (other than to complete this assignment)**

Meta collected this for curated advertising, and to provide curated content recommendations to keep users on Instagram longer. This data is being used for the assignment so we can practice analyzing data and coming up with a hypothesis around it.

In [8]:
separate_likes['string_list_data'][0][0]['timestamp']

1651550611

In [9]:
time = [x[0]['timestamp'] for x in separate_likes['string_list_data']]

In [10]:
timestamp = pd.DataFrame(time)

In [11]:
timestamp

0
0     1651550611
1     1651545320
2     1651545318
3     1651545315
4     1651539832
...          ...
3437  1661366867
3438  1661366733
3439  1661366708
3440  1661366702
3441  1661366698

[3442 rows x 1 columns]

https://www.unixtimestamp.com Use this website to convert timestamp to regular date and times

In [12]:
separate_likes['string_list_data'][0][0]['href']

'https://www.instagram.com/reel/CcBC8qdJ_7W/'

In [13]:
link = [x [0]['href'] for x in separate_likes['string_list_data']]

In [14]:
href = pd.DataFrame(link)

In [15]:
href

0
0     https://www.instagram.com/reel/CcBC8qdJ_7W/
1        https://www.instagram.com/p/CczbkJ9PUx9/
2        https://www.instagram.com/p/Cc4H2pKB-cm/
3        https://www.instagram.com/p/Cc-Cn7IjWGR/
4     https://www.instagram.com/reel/CdDJUu-AnWj/
...                                           ...
3437  https://www.instagram.com/reel/ChaVQSdMrli/
3438     https://www.instagram.com/p/Chmlf7KNxye/
3439  https://www.instagram.com/reel/Cgr06zUITQp/
3440     https://www.instagram.com/p/CghUQG8rWFq/
3441     https://www.instagram.com/p/Chfxr5_PY96/

[3442 rows x 1 columns]

In [16]:
separate_likes['timestamp'] = timestamp

In [17]:
separate_likes['href'] = href

I wanted to add both the timestamp and the link of the original liked post to the full DataFrame

In [18]:
separate_likes

title media_list_data  \
0            kristeacafe              []   
1             rakib.uiux              []   
2         3dgamingsetups              []   
3            websitekana              []   
4          redheadstacyx              []   
...                  ...             ...   
3437      coffeecouple17              []   
3438  michelle.languages              []   
3439         pughieplays              []   
3440        pinkshokupan              []   
3441            funknglo              []   

                                       string_list_data   timestamp  \
0     [{'href': 'https://www.instagram.com/reel/CcBC...  1651550611   
1     [{'href': 'https://www.instagram.com/p/CczbkJ9...  1651545320   
2     [{'href': 'https://www.instagram.com/p/Cc4H2pK...  1651545318   
3     [{'href': 'https://www.instagram.com/p/Cc-Cn7I...  1651545315   
4     [{'href': 'https://www.instagram.com/reel/CdDJ...  1651539832   
...                                                 ...         ...   
3437  [{'href': 'https://www.instagram.com/reel/ChaV...  1661366867   
3438  [{'href': 'https://www.instagram.com/p/Chmlf7K...  1661366733   
3439  [{'href': 'https://www.instagram.com/reel/Cgr0...  1661366708   
3440  [{'href': 'https://www.instagram.com/p/CghUQG8...  1661366702   
3441  [{'href': 'https://www.instagram.com/p/Chfxr5_...  1661366698   

                                             href  
0     https://www.instagram.com/reel/CcBC8qdJ_7W/  
1        https://www.instagram.com/p/CczbkJ9PUx9/  
2        https://www.instagram.com/p/Cc4H2pKB-cm/  
3        https://www.instagram.com/p/Cc-Cn7IjWGR/  
4     https://www.instagram.com/reel/CdDJUu-AnWj/  
...                                           ...  
3437  https://www.instagram.com/reel/ChaVQSdMrli/  
3438     https://www.instagram.com/p/Chmlf7KNxye/  
3439  https://www.instagram.com/reel/Cgr06zUITQp/  
3440     https://www.instagram.com/p/CghUQG8rWFq/  
3441     https://www.instagram.com/p/Chfxr5_PY96/  

[3442 rows x 5 columns]

**c) In what ways may this data be reliable?**

This data is reliable because it comes from Instagram by request through their website. The information Instagram provided in the like json file is good for finding very old liked post.

In [19]:
final_like_data = separate_likes.drop(columns = ['string_list_data', 'media_list_data'])

In [20]:
final_like_data

title   timestamp  \
0            kristeacafe  1651550611   
1             rakib.uiux  1651545320   
2         3dgamingsetups  1651545318   
3            websitekana  1651545315   
4          redheadstacyx  1651539832   
...                  ...         ...   
3437      coffeecouple17  1661366867   
3438  michelle.languages  1661366733   
3439         pughieplays  1661366708   
3440        pinkshokupan  1661366702   
3441            funknglo  1661366698   

                                             href  
0     https://www.instagram.com/reel/CcBC8qdJ_7W/  
1        https://www.instagram.com/p/CczbkJ9PUx9/  
2        https://www.instagram.com/p/Cc4H2pKB-cm/  
3        https://www.instagram.com/p/Cc-Cn7IjWGR/  
4     https://www.instagram.com/reel/CdDJUu-AnWj/  
...                                           ...  
3437  https://www.instagram.com/reel/ChaVQSdMrli/  
3438     https://www.instagram.com/p/Chmlf7KNxye/  
3439  https://www.instagram.com/reel/Cgr06zUITQp/  
3440     https://www.instagram.com/p/CghUQG8rWFq/  
3441     https://www.instagram.com/p/Chfxr5_PY96/  

[3442 rows x 3 columns]

I used the .drop method to remove the string_list_data column since I already had that information on the DataFrame split into different columns. And I dropped media_list_data as all the values were the same. 

_This is DataFrame is the most sensible to me_

**d) In what ways may this data be unreliable?**

The data is outdated, unless a person specifically needs data from when I created my account on May 3 to August 24, this data is really old because I've liked a lot of posts since then. Another thing is that, some posts could of been liked unintentionally, so this set of data might not perfectly reflect me and my interest. Also, this data only provides the original poster, a link to the post, and time it was liked which may not be useful for all research.

In [21]:
final_like_data.count()

title        3442
timestamp    3442
href         3442
dtype: int64

I now know that I've liked 3,442 posts

In [22]:
top_liked = final_like_data.groupby(['title']).count().sort_values('href')

In [23]:
top_liked

timestamp  href
title                                
zzaytrippin                   1     1
matchanomena                  1     1
matchamondayspodcast          1     1
matcha_trip                   1     1
unoteahouse                   1     1
...                         ...   ...
kentstate                    39    39
dunkin                       43    43
hellokitty                   50    50
positivelypresent            57    57
starbucks                    64    64

[1745 rows x 2 columns]

Grouping by the total sum of titles removes the repeated titles (or original Instagram users)

I sorted the values by the href, so the values would be ordered in the proper numerical order instead of the title's alphabetical order. Now I am able to see which Instagram account I liked the most and how many unique Instagram accounts I've interacted with.

By converting the Timestamps to regular date and time format, I learned that between May 3 at 12:03 am and August 24 at 2:44 pm I've liked a total of 3,442 Instagram post

From grouping and sorting I learned that I liked 64 posts by Starbucks, making Starbucks my most liked account. I also know that I've liked posts from 1,745 different Instagram accounts

**After viewing the other data files provided by Instagram, state a hypothesis and a data file that may help answer the hypothesis.**

**a) Make sure you explain the theoretical hypothesis and the statistical hypothesis.**
Based of the data received I spent 113 days, or 3 months and 21 days, on Instagram and liked 3,442 Instagram post, I could estimate that I liked around 30 post a day, from various accounts that weren't only by the accounts I follow. If I liked 3,442 post within 113 days, then I spend too much time on Instagram.

**b) Instead of a data file provided by Instagram, what would be the idea data to use to test your hypothesis?**
Apple has a feature called Screen Time that tracks how long you've used an app daily or weekly, but that only keeps information for 4 weeks, and Instagram has a feature on the app itself that tracks how long you're on the app for, but it only keeps information for 7 days. If there is a way to receive past data from Apple or Instagram I could prove that I spent to much time on Instagram. I would also need to find out how long the average person spends time on Instagram to compare my using habits to see if I'm above or below the average. 

# Likes Report
## Tiffany Marks
## September 20, 2022

On September 6, I downloaded my Instagram data for this report. I know that at the time I deleted Instagram off my phone so I could focus more on school work. So most of the information I received stops around August 24, the day before school started. First I renamed pandas to pd and imported json into the likes_Report.ipynb file. Next, I downloaded the liked_post.json file and used json load function to load the data into the notebook. 

I used pd.DataFrame() to convert the json file data into a DataFrame. I separated the data given into columns to make the data more readable. I used pd.DataFrame() to split the content between the brackets of media_likes_media{} into 3 separate columns.

In [24]:
pd.DataFrame(insta_likes['likes_media_likes'])

title media_list_data  \
0            kristeacafe              []   
1             rakib.uiux              []   
2         3dgamingsetups              []   
3            websitekana              []   
4          redheadstacyx              []   
...                  ...             ...   
3437      coffeecouple17              []   
3438  michelle.languages              []   
3439         pughieplays              []   
3440        pinkshokupan              []   
3441            funknglo              []   

                                       string_list_data  
0     [{'href': 'https://www.instagram.com/reel/CcBC...  
1     [{'href': 'https://www.instagram.com/p/CczbkJ9...  
2     [{'href': 'https://www.instagram.com/p/Cc4H2pK...  
3     [{'href': 'https://www.instagram.com/p/Cc-Cn7I...  
4     [{'href': 'https://www.instagram.com/reel/CdDJ...  
...                                                 ...  
3437  [{'href': 'https://www.instagram.com/reel/ChaV...  
3438  [{'href': 'https://www.instagram.com/p/Chmlf7K...  
3439  [{'href': 'https://www.instagram.com/reel/Cgr0...  
3440  [{'href': 'https://www.instagram.com/p/CghUQG8...  
3441  [{'href': 'https://www.instagram.com/p/Chfxr5_...  

[3442 rows x 3 columns]

I did have trouble splitting the string_list_data content into separate columns. I had the most problems with this part of the project because I tried figuring out how to do it before we learned about the x for x method we learned in class. I used the x for x method we used in class to split the content between the string_list_data brackets and curly brackets. After separating the string_list_data content, I add them to the original DataFrame.

In [25]:
separate_likes

title media_list_data  \
0            kristeacafe              []   
1             rakib.uiux              []   
2         3dgamingsetups              []   
3            websitekana              []   
4          redheadstacyx              []   
...                  ...             ...   
3437      coffeecouple17              []   
3438  michelle.languages              []   
3439         pughieplays              []   
3440        pinkshokupan              []   
3441            funknglo              []   

                                       string_list_data   timestamp  \
0     [{'href': 'https://www.instagram.com/reel/CcBC...  1651550611   
1     [{'href': 'https://www.instagram.com/p/CczbkJ9...  1651545320   
2     [{'href': 'https://www.instagram.com/p/Cc4H2pK...  1651545318   
3     [{'href': 'https://www.instagram.com/p/Cc-Cn7I...  1651545315   
4     [{'href': 'https://www.instagram.com/reel/CdDJ...  1651539832   
...                                                 ...         ...   
3437  [{'href': 'https://www.instagram.com/reel/ChaV...  1661366867   
3438  [{'href': 'https://www.instagram.com/p/Chmlf7K...  1661366733   
3439  [{'href': 'https://www.instagram.com/reel/Cgr0...  1661366708   
3440  [{'href': 'https://www.instagram.com/p/CghUQG8...  1661366702   
3441  [{'href': 'https://www.instagram.com/p/Chfxr5_...  1661366698   

                                             href  
0     https://www.instagram.com/reel/CcBC8qdJ_7W/  
1        https://www.instagram.com/p/CczbkJ9PUx9/  
2        https://www.instagram.com/p/Cc4H2pKB-cm/  
3        https://www.instagram.com/p/Cc-Cn7IjWGR/  
4     https://www.instagram.com/reel/CdDJUu-AnWj/  
...                                           ...  
3437  https://www.instagram.com/reel/ChaVQSdMrli/  
3438     https://www.instagram.com/p/Chmlf7KNxye/  
3439  https://www.instagram.com/reel/Cgr06zUITQp/  
3440     https://www.instagram.com/p/CghUQG8rWFq/  
3441     https://www.instagram.com/p/Chfxr5_PY96/  

[3442 rows x 5 columns]

After going through the data, using the drop method to remove columns, and organizing it into a DataFrame, I had a DataFrame that made the most sense for my use. For my most sensible DataFrame, I chose to keep the following information: The Title, Timestamp and href. The title reveals the name of the account that created the post, the Timestamp reveals when the post was liked once translated from the Timestamp format and the href provides a direct link to the exact post. Information removed didn't provide me information I understood or was just repetitive.

In [26]:
final_like_data

title   timestamp  \
0            kristeacafe  1651550611   
1             rakib.uiux  1651545320   
2         3dgamingsetups  1651545318   
3            websitekana  1651545315   
4          redheadstacyx  1651539832   
...                  ...         ...   
3437      coffeecouple17  1661366867   
3438  michelle.languages  1661366733   
3439         pughieplays  1661366708   
3440        pinkshokupan  1661366702   
3441            funknglo  1661366698   

                                             href  
0     https://www.instagram.com/reel/CcBC8qdJ_7W/  
1        https://www.instagram.com/p/CczbkJ9PUx9/  
2        https://www.instagram.com/p/Cc4H2pKB-cm/  
3        https://www.instagram.com/p/Cc-Cn7IjWGR/  
4     https://www.instagram.com/reel/CdDJUu-AnWj/  
...                                           ...  
3437  https://www.instagram.com/reel/ChaVQSdMrli/  
3438     https://www.instagram.com/p/Chmlf7KNxye/  
3439  https://www.instagram.com/reel/Cgr06zUITQp/  
3440     https://www.instagram.com/p/CghUQG8rWFq/  
3441     https://www.instagram.com/p/Chfxr5_PY96/  

[3442 rows x 3 columns]

After creating the most sensible DataFrame, I used the groupby() and count() DataFrame methods to organize my data and learn more from what I collected. I know I liked 3,442 posts in total, Starbucks is the account I've liked the most and I've liked posts from 1,745 separate accounts. 

In [27]:
final_like_data.count()

title        3442
timestamp    3442
href         3442
dtype: int64

In [28]:
top_liked

timestamp  href
title                                
zzaytrippin                   1     1
matchanomena                  1     1
matchamondayspodcast          1     1
matcha_trip                   1     1
unoteahouse                   1     1
...                         ...   ...
kentstate                    39    39
dunkin                       43    43
hellokitty                   50    50
positivelypresent            57    57
starbucks                    64    64

[1745 rows x 2 columns]

From the data I collected, I came up with this hypothesis: If I liked 3,442 posts within 113 days, then I spend too much time on Instagram. Other than Instagram's files, my plan is to use data from Apple's Screen Time feature or Instagram's own in-app tracking tool to prove my hypothesis. I also want to compare my data with others to see how many posts they've liked within May and August.